src/main/resources/twilio-csharp/Pagination.mustache

```diff
@@ -3,11 +3,11 @@
         /// <param name="targetUrl"> API-generated URL for the requested results page </param>
         /// <param name="client"> Client to make requests to Twilio </param>
         /// <returns> The target page of records </returns>
-        public static Page<{{apiName}}{{resourceConstant}}> GetPage(string targetUrl, ITwilioRestClient client)
+        public static Page<{{apiName}}{{resourceConstant}}> GetPage(string targetUrl, {{clientName}} client)
         {
-            client = client ?? TwilioClient.GetRestClient();
+            client = client ?? Twilio{{domainClassPrefix}}Client.Get{{restClientClassName}}RestClient();
 
-            var request = new Request(
+            var request = new {{requestName}}Request(
```

Comment:
Do we need to have different request model ?
What if we use AuthStrategy class as discussed.
AuthStrategy can have child classes which will handle different types of Authentication mechanism
For Example
BasicAuthStrategy, TokenAuthStrategy, NoAuthStrategy.